syntax = "proto3";
package waddlebot.workflow;

service WorkflowService {
    rpc TriggerWorkflow(TriggerWorkflowRequest) returns (SuccessResponse);
}

message TriggerWorkflowRequest {
    string token = 1;
    string workflow_id = 2;
    string trigger_source = 3;
    string trigger_data = 4;
    string session_id = 5;
    string entity_id = 6;
    int32 user_id = 7;
    string platform = 8;
}

message SuccessResponse {
    bool success = 1;
    string message = 2;
    string error = 3;
}
